test/wrapper_array.jl

```diff
@@ -52,6 +52,18 @@ end
         @test view(nda, 1,1) == view(nda.data, 1,1)
     end
 
+    @testset "No arguments" begin
+        ndm = NamedDimsArray([10 20; 30 40], (:x, :y))
+        ndv = NamedDimsArray([1, 2, 3], (:x,))
+        @test_throws BoundsError ndm[]
+        @test_throws BoundsError ndv[]
+
+        nds = NamedDimsArray([4], (:x,))  # 1d 1el array (vector with one element)
+        nds2 = NamedDimsArray{()}(fill(4)); # 0d 1el array (scalar)
+        @test nds[] == 4
+        @test nds2[] == 4
```

Comment:
The general pattern for tests is:
*arrange-act-assert*, **arrange-act-assert*, *arrange-act-assert*.
Rather than:
**arrange-arrange-arrange, act-act-act, assert-assert-assert**
so as to minimize the distances between the setup and the assertion.

some times we violate that a bit by declaring a dome things at the start and reusing them for many assertions.
but here we use each dclared variable once.
So we can do declare-test declare-test
and minimise distance we need to do that context over

Reply:
Thanks for the added explanation, these are really helpful!